Digits for renewable clusters' time series

guilpier-code opened the pull request:
Because the **matrix of data series** cannot be of same type for thermal and renewable clusters, we have to split it in 2 specialized parts.
Actually, there is a minimum common part that remains.
This split has consequences in cluster and cluster lists classes of both types (thermal and renewable).

flomnes committed:
Add TimeSeriesMode to renewable clusters

flomnes committed:
Add getter/setter for RenewableCluster::tsMode

flomnes committed:
Ren. clusters : read/write unit count, nom. capacity, ts-interpretation

flomnes committed:
RenewableCluster : use nominalCapacity according to tsMode

flomnes committed:
Update submodule Antares_Simulator_Tests

flomnes committed:
Merge branch 'feature/renewable' into feature/renewable-ts-interpretation

guilpier-code committed:
Digits for renewable clusters : data series off the common series class and consequences

Because the matrix of data series cannot be of same type for thermal and renewable clusters, we have to split it in 2 specialized parts.
Actually, there minimum common part remains.
This split has consequences in cluster and cluster lists classes of both types (thermal and renewable).

guilpier-code committed:
Digits for renewable clusters : bug fix (clusters constructors & series)

flomnes commented on src/libs/antares/study/parts/common/cluster_list.h:
```diff
@@ -8,6 +8,19 @@
+        uint32 operator()(uint32 value) const
+        {
+            return value + 1;
+        }
```
Defining functions in header files usually involves multiple definition issues at link time.

One solution is to use the `inline` keyword to tell the compiler not to create a symbol.

A better solution is to place function declarations in headers and function definitions in source (.cpp) files.

guilpier-code committed:
Digits for renewable clusters : a bit of cleaning

main things :
- RenewableClusterList::loadFromFolder takes the study as argument only because of its study.buffer.
  Clearly, we not need the study for this.
  They are consequences to removing the study as argument : function loadFromFolder(...) cannot override a pure virtual function in the common cluster list class.
  As the ThermalClusterList::loadFromFolder(...) needs study as argument, we remove the common pure virtual function.
- class RenewableCluster : some function are clearly useless in the context of renewable cluster, they are an old legacy from thermal clusters
  We remove them.

flomnes commented on src/libs/antares/study/parts/renewable/cluster.h:
```diff
@@ -167,10 +178,17 @@ class RenewableCluster final : public Cluster
-    // TODO remove
-    void* prepro;
+    bool setTimeSeriesModeFromString(const YString& value);
+    int saveDataSeriesToFolder(const AnyString& folder) const;
+    int loadDataSeriesFromFolder(Study& s, const AnyString& folder);
```
Mark overriding methods as `override`

flomnes commented on src/libs/antares/study/parts/thermal/cluster_list.cpp:
```diff
@@ -712,6 +700,111 @@ void ThermalClusterList::enableMustrunForEveryone()
+    }
+}
+
+bool ThermalClusterList::storeTimeseriesNumbers(Study& study)
+{
+    if (cluster.empty())
+        return true;
+
+    bool ret = true;
+    TSNumbersPredicate predicate;
+
+    each([&](const ThermalCluster& cluster) {
+        study.buffer = study.folderOutput;
+        study.buffer << SEP << "ts-numbers" << SEP << typeID() << SEP << cluster.parentArea->id
+            << SEP << cluster.id() << ".txt";
+        ret = cluster.series->timeseriesNumbers.saveToCSVFile(study.buffer, 0, true, predicate)
+            and ret;
+    });
+    return ret;
+}
```
this method can be placed in ClusterList, `typeID()`, `cluster` all make sense there.

flomnes reviewed (changes requested):
A lot of unnecessary code re-duplication

flomnes commented:
Instead of re-duplicating code, I suggest introducing a pure virtual function for each cluster type
```cpp
virtual unsigned int precision() = 0;
```
This function would return 0 in class `ThermalClusterList` and e.g 4 in `RenewableClusterList`.

guilpier-code committed:
Digits for renewable clusters :  reparing a cmake error

flomnes committed:
Add unit tests for matrices

To check that Yuni::Matrix<double, double> behaves as expected :
- For thermal TS, read/write integers, store them as double
- For renewable TS, read/write doubles with precision=4

flomnes committed:
Revert "Digits for renewable clusters : data series off the common series class and consequences"

This reverts commit 2a306ab8ee12355b71d8a19ae448f4ebb435f7d7.

flomnes committed:
Use precision = 0 for thermal TS, precision = 4 for renw TS

flomnes committed:
Remove unused RenewableCluster constructor

flomnes committed:
Apply clang-format

flomnes committed:
Create & use RenewableCluster::valueAtTimeStep

to reduce "code smells".

flomnes committed:
Merge branch 'feature/renewable-ts-interpretation' into feature/digits_for_renew_cluster_ts

flomnes committed:
Remove 0.d

VC doesn't like it

D:\a\Antares_Simulator\Antares_Simulator\src\libs\antares\study\parts\renewable\cluster.cpp(269,12): error C3688: invalid literal suffix 'd'; literal operator or literal operator template 'operator ""d' not found [D:\a\Antares_Simulator\Antares_Simulator\_build\libs\antares\libantares-core.vcxproj]

flomnes committed:
Merge branch 'feature/renewable-ts-interpretation' into feature/digits_for_renew_cluster_ts

flomnes committed:
PR remarks

* Remove unused prepro attribute, specialize loadDataSeriesFromFolder
method
* Correct value when tsMode == powerGeneration

flomnes committed:
PR remarks

* Remove unused prepro attribute, specialize loadDataSeriesFromFolder
method
* Correct value when tsMode == powerGeneration

flomnes committed:
Revert "PR remarks"

Wrong branch

This reverts commit f6ae44830b52a47acfb0441f89432f73a554fc3e.

flomnes committed:
Merge remote-tracking branch 'origin/feature/renewable-ts-interpretation' into feature/digits_for_renew_cluster_ts

flomnes committed:
Fix renewable prod. output

Take tsMode switch into account

flomnes committed:
Merge branch 'feature/renewable-ts-interpretation' into feature/digits_for_renew_cluster_ts

flomnes committed:
getTimeSeriesModeAsString() returns a YString

flomnes committed:
Merge pull request #272 from AntaresSimulatorTeam/feature/renewable-ts-interpretation

Add gen. modulation interpretation for renewable TS
